java/src/jmri/jmrix/dccpp/swing/DCCppRosterExportAction.java

```diff
@@ -0,0 +1,58 @@
+package jmri.jmrix.dccpp.swing;
+
+import java.awt.event.ActionEvent;
+import javax.swing.AbstractAction;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import javax.script.Bindings;
+import javax.script.ScriptException;
+import javax.script.SimpleBindings;
+
+import jmri.JmriException;
+import jmri.script.ScriptEngineSelector;
+
+/**
+ * Menu action to Export all JMRI Roster Entries as ROSTER() macro calls
+ *  This adds a menu item to the "DCC++" menu. Taking this option opens a 
+ *  window where the ROSTER() lines can be copied into your myAutomation.h 
+ *  Syntax: ROSTER(999,"Loco Name","F0/F1/*F2/F3/F4/F5/F6/F7/F8")
+ * NOTE: calls existing script for compatibility
+ * @author MSteveTodd Copyright (C) 2023
+ */
+public class DCCppRosterExportAction extends AbstractAction {
+
+    private final ScriptEngineSelector _scriptEngineSelector = new ScriptEngineSelector();
+    private final String scriptFilename = "program:jython"+File.separator+"DCC-EX"+File.separator+"RosterExportToDCC-EX.py";
+    private static final Logger log = LoggerFactory.getLogger(DCCppRosterExportAction.class);
+    
+    public DCCppRosterExportAction(String s) {
+        super(s);
+    }
+
+    public DCCppRosterExportAction() {
+        this("Roster Export to DCC-EX");
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        log.debug("in actionPerformed");
+        ScriptEngineSelector.Engine engine =
+                    _scriptEngineSelector.getSelectedEngine();
+        if (engine == null) {
+            log.error("Script engine is null");
```

Comment:
A `JOptionPane.showMessageDialog()` would be useful here. If this error happens now, JMRI will not tell the user what's wrong, unless the user looks in the system console.

https://docs.oracle.com/javase/tutorial/uiswing/components/dialog.html

```
//custom title, error icon
JOptionPane.showMessageDialog(frame,
    "Eggs are not supposed to be green.",
    "Inane error",
    JOptionPane.ERROR_MESSAGE);
```